pegasus/sites.v3/code.org/public/js/volunteer-local.js

```diff
@@ -0,0 +1,325 @@
+var gmap;
+var gmap_loc;
+
+$(function() {
+  selectize = $('#volunteer-search-facets select').selectize({
+    plugins: ['fast_click']
+  });
+
+  $("#location").geocomplete()
+    .bind("geocode:result", function(event, result){
+      var loc = result.geometry.location;
+      gmap_loc = loc.lat() + ',' + loc.lng();
+      resetFacets();
+      submitForm();
+    });
+
+  // Make the map sticky.
+  $("#gmap").sticky({topSpacing:0});
+
+  // Trigger query when a facet is changed.
+  $('#volunteer-search-facets').find('select').change(function() {
+    submitForm();
+  });
+});
+
+function submitForm() {
+  var form_data = $('#volunteer-search-form').serializeArray();
+
+  // Clear the location details.
+  $('#location-details').html('');
+
+  // If we still don't have coordinates, display an error.
+  if (!gmap_loc) {
+    displayQueryError();
+    return;
+  }
+
+  var params = getParams(form_data);
+  sendQuery(params);
+}
+
+function getLatLng(address) {
+  var geocoder = new google.maps.Geocoder();
+
+  geocoder.geocode({'address': address}, function(results, status) {
+    if (status == google.maps.GeocoderStatus.OK) {
+      var loc;
+      loc = results[0].geometry.location;
+      gmap_loc = loc.d + ',' + loc.e;
+    } else {
+      displayQueryError();
+    }
+  });
+}
+
+function getParams(form_data) {
+  var params = [];
+
+  params.push({
+    name: 'coordinates',
+    value: gmap_loc
+  });
+
+  $.each(form_data, function(key, field) {
+    if (field.value !== '' && field.name != 'location') {
+      params.push(field);
+    }
+  });
+
+  return params;
+}
+
+function sendQuery(params) {
+  $.post('/forms/VolunteerEngineerSubmission2015/query', $.param(params), function(response){
+    var results = JSON.parse(response); // Convert the JSON string to a JavaScript object.
+    var locations = getLocations(results);
+    updateResults(locations);
+    updateFacets(results);
+  }).fail(displayQueryError);
+}
+
+function updateResults(locations) {
+  if (locations.length > 0) {
+    $('#volunteer-search-facets').show();
+    $('#controls').html('');
+  } else {
+    displayNoResults();
+  }
+  $('#volunteer-search-results').show();
+
+  loadMap(locations);
+}
+
+function getLocations(results) {
+  var locations = [];
+
+  if(results.response){
+    var volunteers = results.response.docs; // The actual volunteers that were returned by Solr.
+    var volunteers_count = volunteers.length;
+
+    for(var i = 0; i < volunteers_count; i++){
+      var index = i;
```

Comment:
it looks like you could just use `i` here (or use `index` in your for loop)? Or is there a reason these need to be different?